src/t8_schemes/t8_standalone/t8_standalone_implementation.hxx

```diff
@@ -0,0 +1,1572 @@
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#ifndef T8_STANDALONE_IMPLEMENTATION_HXX
+#define T8_STANDALONE_IMPLEMENTATION_HXX
+
+#include <t8_schemes/t8_scheme.hxx>
+#include <t8_eclass.h>
+#include <sc_functions.h>
+#include <t8_schemes/t8_standalone/t8_standalone_elements.hxx>
+
+template <t8_eclass TEclass>
+struct t8_standalone_scheme
+{
+ public:
+  /** Constructor
+   * \param [in] elem_size  The size of the elements this scheme holds.
+  */
+  t8_standalone_scheme ()
+    : element_size (sizeof (t8_standalone_element<TEclass>)), scheme_context (sc_mempool_new (element_size)) {};
+
+ protected:
+  size_t element_size;  /**< The size in bytes of an element of class \a eclass */
+  void *scheme_context; /**< Anonymous implementation context. */
+
+ public:
+  /** Destructor for all default schemes */
+  ~t8_standalone_scheme ()
+  {
+    T8_ASSERT (scheme_context != NULL);
+    SC_ASSERT (((sc_mempool_t *) scheme_context)->elem_count == 0);
+    sc_mempool_destroy ((sc_mempool_t *) scheme_context);
+  }
+
+  /** Move constructor */
+  t8_standalone_scheme (t8_standalone_scheme &&other) noexcept
+    : element_size (other.element_size), scheme_context (other.scheme_context)
+  {
+    other.scheme_context = nullptr;
+  }
+
+  /** Move assignment operator */
+  t8_standalone_scheme &
+  operator= (t8_standalone_scheme &&other) noexcept
+  {
+    if (this != &other) {
+      // Free existing resources of moved-to object
+      if (scheme_context) {
+        sc_mempool_destroy ((sc_mempool_t *) scheme_context);
+      }
+
+      // Transfer ownership of resources
+      element_size = other.element_size;
+      scheme_context = other.scheme_context;
+
+      // Leave the source object in a valid state
+      other.scheme_context = nullptr;
+    }
+    return *this;
+  }
+
+  /** Copy constructor */
+  t8_standalone_scheme (const t8_standalone_scheme &other)
+    : element_size (other.element_size), scheme_context (sc_mempool_new (other.element_size)) {};
+
+  /** Copy assignment operator */
+  t8_standalone_scheme &
+  operator= (const t8_standalone_scheme &other)
+  {
+    if (this != &other) {
+      // Free existing resources of assigned-to object
+      if (scheme_context) {
+        sc_mempool_destroy ((sc_mempool_t *) scheme_context);
+      }
+
+      // Copy the values from the source object
+      element_size = other.element_size;
+      scheme_context = sc_mempool_new (other.element_size);
+    }
+    return *this;
+  }
+
+  // ################################################____GENERAL INFO____################################################
+
+  /** Return the tree class of this scheme.
+   * \return The tree class of this scheme.
+   */
+  constexpr t8_eclass_t
+  get_eclass (void) const
+  {
+    return TEclass;
+  }
+
+  constexpr size_t
+  get_element_size (void) const noexcept
+  {
+    return element_size;
+  }
```

Comment:
I think we could remove `element_size` from the class and replace it here by `sizeof (t8_standalone_element<TEclass>)`
Then we can make this function static as well

Reply:
I like this idea, too.